lib/turn.rb

```diff
@@ -0,0 +1,163 @@
+require './lib/card'
+require './lib/deck'
+require './lib/player'
+require 'pry'
+
+class Turn
+
+    attr_reader :player1,
+                :player2,
+                :spoils_of_war
+    
+    def initialize(player1, player2)
+
+        @player1 = player1
+        @player2 = player2
+        @spoils_of_war = []
+        
+    end
+
+    def type
+        
+        if @player1.deck.rank_of_card_at(0) != @player2.deck.rank_of_card_at(0)
+            :basic
+        elsif @player1.deck.rank_of_card_at(0) == @player2.deck.rank_of_card_at(0) 
+
+            if not_enough_cards? == FALSE
+
+                if @player1.deck.rank_of_card_at(2) != @player2.deck.rank_of_card_at(2)
+                :war
+                else
+                :mutually_assured_destruction
+                end
+
+            else
+
+                :war
+                
+            end
+            
+
+        end
+    end
+
+    def winner
+
+
+        if type == :basic
+
+                
+            if @player1.deck.rank_of_card_at(0) > @player2.deck.rank_of_card_at(0)
+            @player1
+            else
+            @player2
+            end
+
+        elsif type == :war
+            
+            if not_enough_cards? == FALSE
+
+                if @player1.deck.rank_of_card_at(2) > @player2.deck.rank_of_card_at(2)
+                @player1
+                else
+                @player2
+                end
+
+            else
+
+            not_enough_cards
+
+            end
+
+        end
+
+    end
+
+
+    def not_enough_cards?
+
+        if @player1.deck.cards.length < 3 
+            TRUE
+        elsif @player1.deck.cards.length < 3 
+            TRUE
+        elsif @player2.deck.cards.length < 3 
+            TRUE
+        elsif @player2.deck.cards.length < 3 
+            TRUE
+        else
+            FALSE
+        end
+
+    end
+
+    def not_enough_cards
```

Comment:
This method name is a little confusing to me, especially with `not_enough_cards?` just above it. I think I'd rename it something like `winner_when_not_enough_cards` since what it is actually doing is returning us a winner.